src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,32 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+import model.Nationality;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int INCLUDE_START_AND_END_YEAR = 1;
+    private static final int FROM_YEAR = 0;
+    private static final int TO_YEAR = 1;
+    private static final int MIN_AGE_OF_CANDIDATES = 35;
+    private static final int MIN_YEARS_OF_RESIDENCE = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return isEligible(candidate);
+    }
+
+    private boolean isEligible(Candidate candidate) {
+        return candidate.isAllowedToVote()
+                && candidate.getAge() >= MIN_AGE_OF_CANDIDATES
+                && candidate.getNationality() == Nationality.UKR.getValue()
```

Comment:
Create constant for REQUIRED_NATIONALITY and do it like this:
```suggestion
                && candidate.getNationality().equals(REQUIRED_NATIONALITY)
```